src/api/requests.ts

```diff
@@ -1,67 +1,49 @@
+import { EMPTY, ReplaySubject, catchError, mergeMap, switchMap } from "rxjs";
+import { fromFetch } from "rxjs/fetch";
+import { fromPromise } from "rxjs/internal/observable/innerFrom";
 import { IKartLapsPerRun, ILapDataset } from "../models/go-kart-types";
 
-// Get all sessions
-export default function getAllRuns(
-  callback: (allRunsJSON: string[]) => void,
-  errorCallback: (error: Error) => void,
-  finallyCallback?: () => void
-) {
-  fetch(`https://go-kart-api.onrender.com/runs`)
-    .then((response) => response.json())
-    .then((res: string[]) => {
-      if (!res) {
-        errorCallback(new Error("Cannot find Run"));
-      } else {
-        callback(res);
-      }
-    })
-    .catch((error: Error) => errorCallback(error))
-    .finally(() => {
-      if (finallyCallback) finallyCallback();
-    });
-}
+export const signalNewFilenameRequest$ = new ReplaySubject<null>(1);
```

Comment:
I am not certain about this, but the use of <null> here concerns me since you are just also send null to this observable in your laps.ts file with the next. I am finding it hard to see the need for this observable, but it can just be to lack of context.

Reply:
I believe it's null due to the fact that I am passing null in the .next since there is nothing that needs to be passed for this api call to work so the signalNewFilenameRequest$ expects a null to come in.

Reply:
What I am seeing on this is that its currently being used to update `allFilesSubject$` call when the page loads.

You can use type void and use `.next()`